packages/runtime/src/components/_internal/ImplWrapper.tsx

```diff
@@ -52,6 +52,7 @@ const _ImplWrapper = React.forwardRef<HTMLDivElement, ImplWrapperProps>((props,
     return () => {
       apiService.off('uiMethod', handler);
       globalHandlerMap.delete(c.id);
+      delete stateManager.store[c.id];
```

Comment:
Don't put store logic in hanlder's `useEffect`. Create another `useEffect` for store logic.